cpp/src/lance/format/schema.h

```diff
@@ -35,17 +35,20 @@ class Field;
 ///
 class Schema final {
  public:
+  /// Field ID type.
+  using FieldIdType = int32_t;
+
   Schema() = default;
 
   /// Construct Lance Schema from Arrow Schema.
-  Schema(const std::shared_ptr<::arrow::Schema>& schema);
+  explicit Schema(const std::shared_ptr<::arrow::Schema>& schema);
```

Comment:
curious - why do we need to prevent implicit conversion here?

Reply:
This is a C++ style guide , and reported by clang-tidy

https://google.github.io/styleguide/cppguide.html#Implicit_Conversions